pkg/commands/waiter.go

```diff
@@ -59,18 +60,24 @@ func (w *Waiter) wait(ctx context.Context, ob runtime.Object, condition watchToo
 		return err
 	}
 
-	if !done {
-		refable, ok := ob.(kmeta.OwnerRefable)
-		if !ok {
-			return errors.New("unexpected type")
-		}
+	refable, ok := ob.(kmeta.OwnerRefable)
+	if !ok {
+		return errors.New("unexpected type")
+	}
 
-		rv := refable.GetObjectMeta().GetResourceVersion()
-		watchOne := newWatchOneWatcher(ctx, refable, w.dynamicClient)
+	listerWatcherOne := newWatchOneWatcher(ctx, refable, w.dynamicClient)
 
-		ctx, cancel := context.WithTimeout(context.Background(), w.timeout)
-		defer cancel()
-		e, err = watchTools.Until(ctx, rv, watchOne, filterErrors(cfs)...)
+	ctx, cancel := context.WithTimeout(context.Background(), w.timeout)
+	defer cancel()
+
+	if !done {
+		e, err = watchTools.UntilWithSync(ctx,
+			&listerWatcherOne,
+			&unstructured.Unstructured{},
+			func(store cache.Store) (bool, error) {
+				return false, nil
```

Comment:
Could also move the `runChecks` func to here and remove the` if !done` but I think its easier to read with this way